.teamcity/src/main/kotlin/model/FunctionalTestBucketProvider.kt

```diff
@@ -171,43 +173,44 @@ class LargeSubprojectSplitBucket(
             subprojects = listOf(subproject.name),
             enableTestDistribution = false,
             extraParameters = if (include) "-PincludeTestClasses=true -x ${subproject.name}:test" else "-PexcludeTestClasses=true", // Only run unit test in last bucket
-            preBuildSteps = prepareTestClassesStep(testCoverage.os)
+            preBuildSteps = prepareTestClassesStep(testCoverage.os, if (include) INCLUDE else EXCLUDE, classes.map { it.toPropertiesLine() })
         )
+}
+
+enum class TestSplitType {
+    INCLUDE, EXCLUDE
+}
 
-    private fun prepareTestClassesStep(os: Os): BuildSteps.() -> Unit {
-        val testClasses = classes.map { it.toPropertiesLine() }
-        val action = if (include) "include" else "exclude"
-        val unixScript = """
+fun prepareTestClassesStep(os: Os, type: TestSplitType, testClasses: List<String>): BuildSteps.() -> Unit {
+    val action = type.name.lowercase()
```

Comment:
💄 I think it would be better to add this string as a field of the enum instead of using `.lowercase()`.

Reply:
Done.